host/xtest/aes_perf.c

```diff
@@ -587,21 +611,43 @@ int aes_perf_runner_cmd_parser(int argc, char *argv[])
 		} else if (!strcmp(argv[i], "-l")) {
 			NEXT_ARG(i);
 			l = atoi(argv[i]);
+		} else if (!strcmp(argv[i], "-c")) {
+			NEXT_ARG(i);
+			if (!strcasecmp(argv[i], "SM4")) {
+				NEXT_ARG(i);
+				if (!strcmp(argv[i], "-m")) {
```

Comment:
"-m" may not be the argument that right follow "-c SM4" argument.
I think you need to store a local variable for cipher_mode and for base_algo and once the argument parsing loop is completed, check set algo to either AES_xxx or SM4_xxx.